server/app/assets/javascripts/questionBank.ts

```diff
@@ -139,11 +140,25 @@ class QuestionBankController {
     )
     questions.forEach((question) => {
       const questionElement = question as HTMLElement
-      const questionContents = questionElement.innerText
+
+      let questionFilterText
+      if (
+        questionElement.hasAttribute(
+          QuestionBankController.RELEVANT_FILTER_TEXT_DATA_ATTR,
+        )
+      ) {
+        questionFilterText = questionElement.getAttribute(
+          QuestionBankController.RELEVANT_FILTER_TEXT_DATA_ATTR,
+        )
+      } else {
+        questionFilterText = questionElement.innerText
+      }
+
       questionElement.classList.toggle(
         'hidden',
         filterString.length > 0 &&
-          !questionContents.toUpperCase().includes(filterString),
+          (questionFilterText == null ||
```

Comment:
**question:** In what case might this be null?

Reply:
`getAttribute` is documented as possibly returning `null`. Even though I check it first using `hasAttribute`, the compiler doesn't know that and still requires a null check.

We could also rewrite this as:
```
let questionFilterText = questionElement.getAttribute(
          QuestionBankController.RELEVANT_FILTER_TEXT_DATA_ATTR,
        )
if (questionFilterText == null) {
  questionFilterText = questionElement.innerText
}
```

To (I think) remove the null check. Thoughts?

(Typescript doesn't have a `?:` operator like [Kotlin](https://kotlinlang.org/docs/null-safety.html#elvis-operator) right?)

Reply:
Oh got it, makes sense! There is actually something called a nullish coalescing operator that looks at the left side and if its null it will return what is on the right side. So in this case it would be:

```
const questionFilterText = questionElement.getAttribute( QuestionBankController.RELEVANT_FILTER_TEXT_DATA_ATTR, ) ?? questionElement.innerText
```

That seems like probably the best option to me but let me know what you think! 

Here are some examples of it being used: https://github.com/search?q=org%3Aciviform%20%22%3F%3F%22&type=code

Reply:
`??` is exactly what I was hoping for, thanks! Done.